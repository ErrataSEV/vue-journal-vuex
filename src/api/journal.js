
import axios from "axios";

const journalApi = axios.create({
    baseURL: 'https://vue-demos-36557-default-rtdb.firebaseio.com'
})

export default journalApi