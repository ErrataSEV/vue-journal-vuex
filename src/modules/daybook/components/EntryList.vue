<template>
    <div class="entry-list-container">
        <div class="px-2 pt-2 pb-2">
            <input type="text" placeholder="Search Entry" class="form-control">
        </div>

        <div class="entry-scrollarea">
            <h2 v-for="item in 10" :key="item">
                <Entry />
            </h2>
        </div>
    </div>
</template>

<script>
import { defineAsyncComponent } from 'vue';

export default {
    components: {
        Entry: defineAsyncComponent(() => import('./EntryComponent.vue'))
    }
}
</script>

<style lang="scss" scoped>
    // input {
    //     height: px;
    // }

    .entry-list-container {
        border-right: 1px solid #2C3E50;
        height: calc( 100px - 56px );
    }

    .entry-scrollarea {
        height: calc( 100vh - 120px );
        overflow: scroll;
    }
</style>