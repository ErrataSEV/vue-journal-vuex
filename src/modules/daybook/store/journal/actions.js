// export default myAction = async ({ commit }) => {
//
// }

import journalApi from "@/api/journal"

export const loadEntries = async ({ commit }) => {

    const { data } = await journalApi.get('/entries.json')
    const entries = []

    if (!data) {
        commit('setEntries', [])
        return
    }

    for (const id of Object.keys(data)) {
        entries.push({
            id,
            ...data[id]
        })
    }
    
    // TODO: sort desc by date

    commit('setEntries', entries)
}

export const updateEntry = async ({ commit }, entry) => {
    const {id, ...body} = entry
    
    await journalApi.put(`/entries/${id}.json`, body)

    commit('updateEntry', { ...entry })
}

export const createEntry = async ({ commit }, entry) => {
    
    const { data } = await journalApi.post('/entries.json', entry)
    
    commit('addEntry', { id: data.name, ...entry })

    return data.name
}


export const deleteEntry = async ({ commit }, id) => {

    await journalApi.delete(`/entries/${id}.json`)

    commit('removeEntry', id)

}
