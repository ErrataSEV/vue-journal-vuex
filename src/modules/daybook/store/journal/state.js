export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Occaecat nostrud elit et nostrud sit irure. Adipisicing quis dolore incididunt sunt aliqua laborum elit irure eiusmod. Sit elit tempor eu velit minim commodo elit. Magna laborum nostrud dolor ea nulla enim qui do duis qui sunt mollit.',
            picture: null,
        },
        {
            id: new Date().getTime()+100,
            date: new Date().toDateString(),
            text: 'Proident reprehenderit minim reprehenderit adipisicing eiusmod tempor minim velit Lorem eiusmod velit non. Excepteur minim voluptate et do ea. Velit mollit dolor esse enim ea cillum aute.',
            picture: null,
        },
        {
            id: new Date().getTime()+1000,
            date: new Date().toDateString(),
            text: 'Ea reprehenderit reprehenderit incididunt nostrud aute proident. Est cupidatat esse ea nostrud qui enim. Veniam et ad pariatur dolor mollit. Velit consectetur deserunt fugiat dolor ad ex enim ex irure sit ex non. Esse incididunt ad consequat eu officia veniam aliquip qui enim nulla enim duis. Do do mollit minim dolor consectetur sint ad est dolor magna aliqua aliqua. Culpa consequat cupidatat dolore id Lorem duis esse.',
            picture: null,
        }
    ]
})