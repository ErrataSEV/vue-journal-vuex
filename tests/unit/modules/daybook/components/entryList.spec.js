import { createStore } from "vuex"
import { shallowMount } from "@vue/test-utils";

import EntryList from '@/modules/daybook/components/EntryList';
import { journalState } from '../../../mock-data/test-journal-state';
import journal from '@/modules/daybook/store/journal';

const createVuexStore = ( initialState ) => createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }
    }
})

describe('Test in the EntryList', () => {
    const store = createVuexStore(journalState)
    const mockRouter = {
        push: jest.fn()
    }
    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallowMount(EntryList, {
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            }
        })
    })

    test('must call the entriesByTerm and show two entries', () => {

        expect( wrapper.html() ).toMatchSnapshot()
        expect( wrapper.findAll('entry-stub').length ).toBe(2)
    })

    test('must call the getEntriesByTerm and filter using the entries', async () => {
        const input = wrapper.find('input')
        await input.setValue('hard')

        expect(wrapper.findAll('entry-stub').length).toBe(1)
    })

    test('the buttton "New" must redirect to /new', () => {
        wrapper.find('button').trigger('click')
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'entry', params:{ id: 'new' } })
    })
})