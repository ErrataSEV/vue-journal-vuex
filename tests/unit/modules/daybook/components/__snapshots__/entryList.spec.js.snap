// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the EntryList must call the entriesByTerm and show two entries 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" placeholder="Search Entry" class="form-control"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> New Entry </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
